49 eslint problems reported

**Describe the bug**
49 Lint problems (46 errors, 3 warnings) in `pixie-blog` source code.

The majority of the issues are in three categories:
1. Unused imports. Trivial fix.
2. Path resolution. A tweak to the eslint config would most likely fix this.
3. Tiny issues like `==` where there should be a `===`.

**To Reproduce**
Steps to reproduce the behavior:

1. `nvm exec 13 yarn lint`

**Expected behavior**

There should be no lint problems reported when running `yarn lint`.

**Additional context**

Command:
`$ nvm exec 13 yarn lint`

Output:
```
Running node v13.14.0 (npm v6.14.4)
yarn run v1.22.4
$ eslint src ./*.js; git ls-files '**/*.tsx' '**/*.ts' '**/*.js' '**/*.jsx' '**/*.scss' '**/*.py' | xargs -l1 tools/licenses/checker.py -f

.../pixie-blog/src/components/image-zoom-modal.plugin.js
   99:19  warning  Unexpected unnamed function  func-names
  114:20  warning  Unexpected unnamed function  func-names
  119:21  warning  Unexpected unnamed function  func-names

.../pixie-blog/src/components/mainThemeProvider.tsx
  20:10  error  'useEffect' is defined but never used      no-unused-vars
  20:21  error  'useRef' is defined but never used         no-unused-vars
  22:26  error  'useMediaQuery' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/baseComponents.tsx
  26:28  error  Unable to resolve path to module 'components/utils'     import/no-unresolved
  27:18  error  Unable to resolve path to module './code'               import/no-unresolved
  27:18  error  Missing file extension for "./code"                     import/extensions
  29:23  error  Unable to resolve path to module './anchor'             import/no-unresolved
  29:23  error  Missing file extension for "./anchor"                   import/extensions
  30:26  error  Unable to resolve path to module './codeRenderer'       import/no-unresolved
  30:26  error  Missing file extension for "./codeRenderer"             import/extensions
  31:22  error  Unable to resolve path to module './listItem'           import/no-unresolved
  31:22  error  Missing file extension for "./listItem"                 import/extensions
  32:29  error  Unable to resolve path to module './custom-table-cell'  import/no-unresolved
  32:29  error  Missing file extension for "./custom-table-cell"        import/extensions
  33:19  error  Unable to resolve path to module './h-link'             import/no-unresolved
  33:19  error  Missing file extension for "./h-link"                   import/extensions
  34:23  error  Unable to resolve path to module './footnotes'          import/no-unresolved
  34:23  error  Missing file extension for "./footnotes"                import/extensions

.../pixie-blog/src/components/mdxComponents/codeRenderer.tsx
   25:34  error  Unable to resolve path to module '../mainThemeProvider'  import/no-unresolved
   25:34  error  Missing file extension for "../mainThemeProvider"        import/extensions
  101:28  error  Expected '===' and instead saw '=='                      eqeqeq

.../pixie-blog/src/components/mdxComponents/command.tsx
  20:10  error  'Theme' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/custom-table-cell.tsx
  22:21  error  Unable to resolve path to module './parseMd'           import/no-unresolved
  22:21  error  Missing file extension for "./parseMd"                 import/extensions
  35:1   error  This line has a length of 134. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/footnotes.tsx
  39:1  error  This line has a length of 106. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/h-link.tsx
  36:22  error  'TypographyVariant' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/index.tsx
  21:28  error  Unable to resolve path to module './baseComponents'  import/no-unresolved
  21:28  error  Missing file extension for "./baseComponents"        import/extensions
  22:25  error  Unable to resolve path to module './svg'             import/no-unresolved
  22:25  error  Missing file extension for "./svg"                   import/extensions
  23:21  error  Unable to resolve path to module './command'         import/no-unresolved
  23:21  error  Missing file extension for "./command"               import/extensions
  24:19  error  Unable to resolve path to module './quote'           import/no-unresolved
  24:19  error  Missing file extension for "./quote"                 import/extensions
  25:24  error  Unable to resolve path to module './block-quote'     import/no-unresolved
  25:24  error  Missing file extension for "./block-quote"           import/extensions

.../pixie-blog/src/components/mdxComponents/listItem.tsx
  33:1  error  This line has a length of 102. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/parseMd.tsx
  20:1   error  '@mdx-js/runtime' should be listed in the project's dependencies. Run 'npm i -S @mdx-js/runtime' to add it  import/no-extraneous-dependencies
  21:27  error  Unable to resolve path to module './index'                                                                  import/no-unresolved
  21:27  error  Missing file extension for "./index"                                                                        import/extensions

.../pixie-blog/src/components/mdxComponents/quote.tsx
  22:10  error  'Theme' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/svg.tsx
  21:34  error  Unable to resolve path to module '../mainThemeProvider'  import/no-unresolved
  21:34  error  Missing file extension for "../mainThemeProvider"        import/extensions

.../pixie-blog/src/components/theme.js
  150:8  error  Missing trailing comma  comma-dangle

.../pixie-blog/src/pages/404.tsx
  22:10  error  'Theme' is defined but never used  no-unused-vars

✖ 49 problems (46 errors, 3 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -10,6 +10,7 @@
   "extends": [
     "plugin:react/recommended",
     "airbnb-base",
+    "airbnb-typescript/base",
     "airbnb/rules/react"
   ],
   "overrides": [
@@ -64,7 +65,6 @@
   },
   "rules": {
     "import/extensions": 0,
-    "import/no-unresolved": 1,
     "quotes": [
       2,
       "single",

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -1,75 +1,91 @@
 {
-    "root": true,
-    "env": {
-        "browser": true,
-        "es6": true
-    },
-    "ignorePatterns": [
-        "dist/**"
-    ],
-    "extends": [
-        "plugin:react/recommended",
-        "airbnb-base",
-        "airbnb/rules/react"
-    ],
-    "overrides": [
-        {
-            "files": [
-                "*.ts",
-                "*.tsx"
-            ],
-            "extends": [
-                "plugin:@typescript-eslint/recommended"
-            ],
-            "rules": {
-                "@typescript-eslint/explicit-function-return-type": "off",
-                "@typescript-eslint/no-explicit-any": "off"
-            }
-        }
-    ],
-    "globals": {
-        "Atomics": "readonly",
-        "SharedArrayBuffer": "readonly"
-    },
-    "parser": "@typescript-eslint/parser",
-    "parserOptions": {
-        "ecmaFeatures": {
-            "jsx": true
-        },
-        "ecmaVersion": 2018,
-        "sourceType": "module",
-        "project": [
-            "./tsconfig.json"
-        ]
-    },
-    "plugins": [
-        "react",
-        "@typescript-eslint"
-    ],
-    "rules": {
-        "import/extensions": 0,
-        "import/no-unresolved": 0,
-        "quotes": [
-            2,
-            "single",
-            "avoid-escape"
-        ],
-        "import/no-extraneous-dependencies": [
-            "error"
-        ],
-        "react/jsx-filename-extension": "off",
-        "react/prop-types": [
-            2,
-            {
-                "skipUndeclared": true
-            }
-        ],
-        "react/jsx-props-no-spreading": [
-            0
-        ],
-        "jsx-quotes": [
-            "error",
-            "prefer-single"
-        ]
+  "root": true,
+  "env": {
+    "browser": true,
+    "es6": true
+  },
+  "ignorePatterns": [
+    "dist/**"
+  ],
+  "extends": [
+    "plugin:react/recommended",
+    "airbnb-base",
+    "airbnb/rules/react"
+  ],
+  "overrides": [
+    {
+      "files": [
+        "*.ts",
+        "*.tsx"
+      ],
+      "extends": [
+        "plugin:@typescript-eslint/recommended"
+      ],
+      "rules": {
+        "@typescript-eslint/explicit-function-return-type": "off",
+        "@typescript-eslint/no-explicit-any": "off"
+      }
     }
+  ],
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "ecmaVersion": 2018,
+    "sourceType": "module",
+    "project": [
+      "./tsconfig.json"
+    ]
+  },
+  "plugins": [
+    "react",
+    "@typescript-eslint"
+  ],
+  "settings": {
+    "import/resolver": {
+      "node": {
+        "extensions": [
+          ".js",
+          ".jsx",
+          ".ts",
+          ".tsx"
+        ],
+        "moduleDirectory": [
+          "node_modules",
+          "src/"
+        ]
+      }
+    }
+  },
+  "rules": {
+    "import/extensions": 0,
+    "import/no-unresolved": 1,
+    "quotes": [
+      2,
+      "single",
+      "avoid-escape"
+    ],
+    "import/no-extraneous-dependencies": [
+      "error"
+    ],
+    "react/jsx-filename-extension": "off",
+    "react/prop-types": [
+      2,
+      {
+        "skipUndeclared": true
+      }
+    ],
+    "react/jsx-props-no-spreading": [
+      0
+    ],
+    "jsx-quotes": [
+      "error",
+      "prefer-single"
+    ]
+  }
 }

--- a/package.json
+++ b/package.json
@@ -106,6 +106,7 @@
     "eslint-plugin-promise": "^4.2.1",
     "eslint-plugin-react": "^7.18.3",
     "eslint-plugin-react-hooks": "^1.7.0",
+    "eslint-config-airbnb-typescript": "^7.2.1",
     "prettier": "^1.17.1",
     "typescript": "^3.7.5"
   },

--- a/src/components/mdxComponents/baseComponents.tsx
+++ b/src/components/mdxComponents/baseComponents.tsx
@@ -29,7 +29,6 @@ import Pre from './pre';
 import AnchorTag from './anchor';
 import CodeRenderer from './codeRenderer';
 import ListItem from './listItem';
-import CustomTableCell from './custom-table-cell';
 import HLink from './h-link';
 import Footnotes from './footnotes';
 
@@ -62,7 +61,6 @@ export default {
   a: (props: any) => <AnchorTag {...props} />,
   table: (props: any) => <Table {...props} />,
   tr: (props: any) => <TableRow {...props} />,
-  td: ({ align, ...props }) => <CustomTableCell {...props} align={align || undefined} />,
   th: ({ align, ...props }) => <TableCell {...props} align={align || undefined} />,
   tbody: (props: any) => <TableBody {...props} />,
   thead: (props: any) => <TableHead {...props} />,

--- a/src/components/mdxComponents/custom-table-cell.tsx
+++ b/src/components/mdxComponents/custom-table-cell.tsx
@@ -19,7 +19,14 @@
 import withStyles from '@material-ui/core/styles/withStyles';
 import React from 'react';
 import TableCell from '@material-ui/core/TableCell';
-import parseMd from './parseMd';
+import MDX from '@mdx-js/runtime';
+import mdxComponents from './baseComponents';
+
+const parse = (input) => (
+  <MDX components={mdxComponents}>
+    {input}
+  </MDX>
+);
 
 const CustomTableCell = withStyles(() => ({
   td: {
@@ -33,7 +40,7 @@ const CustomTableCell = withStyles(() => ({
   }
   return (
     <TableCell className={classes.td} align={align || undefined}>
-      {parsableChildren ? parseMd(parsableChildren) : children}
+      {parsableChildren ? parse(parsableChildren) : children}
     </TableCell>
   );
 });

--- a/src/components/mdxComponents/index.tsx
+++ b/src/components/mdxComponents/index.tsx
@@ -23,6 +23,7 @@ import SvgRenderer from './svg';
 import Command from './command';
 import Quote from './quote';
 import BlockQuote from './block-quote';
+import CustomTableCell from './custom-table-cell';
 
 export default {
   ...baseComponents,
@@ -30,4 +31,6 @@ export default {
   quote: (props: any) => <Quote {...props} />,
   blockquote: (props: any) => <BlockQuote {...props} />,
   command: (props: any) => <Command {...props} />,
+  td: ({ align, ...props }) => <CustomTableCell {...props} align={align || undefined} />,
+
 };

